### DISCUSSION TITLE
Error while executing procudure with rollback in postgresql database

### DISCUSSION THREAD

**Discussion by 93910676wangd** [dbeaver/dbeaver#16059] [General]
#### System information: 
- DBeaver version:dbeaver-ce-21.3.2-x86_64(windows)

#### Connection specification:
- Additional extensions:postgresql13.3(centos7.9)
PostgreSQL JDBC Driver

#### Describe the problem you're observing:

Executing the following statement with DBeaver tool will cause PostgreSQL DB exception
CALL "TEST_A"();
![error](https://user-images.githubusercontent.com/19505572/150262053-089fafff-108d-4ec1-b238-fd5a58c6e531.png)
DB log：**the database system is in recovery mode**

~~~sql
CREATE OR REPLACE PROCEDURE public."TEST_A"()
 LANGUAGE plpgsql
AS $procedure$
 DECLARE
	CCPG_SaiSeion$kanaReplaceArray_  varchar ARRAY[5];
	i numeric := 0;
	j numeric := 0;
	lsql varchar;
	lcsrToti refcursor := 'curs';
	a varchar;
	lProcCount integer := 0;
	lHjnHyokagak numeric;
	lHjnZenHyokagak numeric;
BEGIN
   
		    FOR i IN 1..10 LOOP
						BEGIN
		            INSERT INTO test values('qq','ww');
				        lProcCount := lProcCount + 1;
				    EXCEPTION
                WHEN NO_DATA_FOUND THEN
                   lHjnHyokagak    := 0;
                   lHjnZenHyokagak := 0;
									 WHEN OTHERS THEN
									     RAISE;
						END;
		
		        IF MOD(lProcCount, /*EEPG_Cnst.CMTCNT_EER02420*/2) = 0 THEN
		            j := j + 1;
				        raise notice'j%',j;
                ROLLBACK;
            END IF;
		END LOOP;
END;
$procedure$
;
~~~

the rollback statement will cause exceptions

However, other tools such as Navicat or pgadmin4 will not introduce exceptions

It won't work to execute commands directly with psql, so it is judged that it is a bug in the DBeaver. Please confirm and fix it

#### Include any warning/errors/backtraces from the logs

postgresql db log: **the database system is in recovery mode*

2022-01-19 17:21:30.840 CST [29227] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.851 CST [29228] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.862 CST [29229] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.924 CST [29237] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.930 CST [29240] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.936 CST [29242] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.942 CST [29245] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.954 CST [29249] FATAL:  データベースシステムはリカバリモードです
2022-01-19 17:21:30.961 CST [29252] FATAL:  データベースシステムはリカバリモードです

**Comment by LonwoLonwo** [comment]
Hello @93910676wangd 

Unfortunately can't reproduce your issue. 
And I can't look into this issue because the issue is in your database.
Database logs before `the database system is in recovery mode` can be helpful.

Maybe this topic can help: https://dba.stackexchange.com/questions/61650/postgres-is-stuck-in-recovery-mode

**Comment by 93910676wangd** [comment]
> Hello @93910676wangd
> 
> Unfortunately can't reproduce your issue. And I can't look into this issue because the issue is in your database. Database logs before `the database system is in recovery mode` can be helpful.
> 
> Maybe this topic can help: https://dba.stackexchange.com/questions/61650/postgres-is-stuck-in-recovery-mode

why  Navicat or pgadmin4 tool is no problem
however DBeaver is error.. why???
